taranis/analyze_schema.py

```diff
@@ -43,70 +60,108 @@ def __init__(
+                            else:
+                                a_quality[record.id]["reason"] = str(e).replace("'", "")
+                            a_quality[record.id]["quality"] = "Bad quality"
                     else:
-                        record_sequence = str(record.seq)
-                    a_quality[record.id]["order"] = sequence_order
-                    if record_sequence[0:3] not in taranis.utils.START_CODON_FORWARD:
                         a_quality[record.id]["quality"] = "Bad quality"
-                        a_quality[record.id]["reason"] = "Start codon not found"
-                    elif record_sequence[-3:] not in taranis.utils.STOP_CODON_FORWARD:
-                        a_quality[record.id]["quality"] = "Bad quality"
-                        a_quality[record.id]["reason"] = "Stop codon not found"
-                    
-                    elif taranis.utils.find_multiple_stop_codons(record_sequence):
-                        a_quality[record.id]["quality"] = "Bad quality"
-                        a_quality[record.id]["reason"] = "Multiple stop codons found"
+                        a_quality[record.id]["reason"] = str(e).replace("'", "")
+
                 if (
                     self.remove_no_cds
                     and a_quality[record.id]["quality"] == "Bad quality"
                 ):
                     bad_quality_record.append(record.id)
 
-        if self.remove_duplicated:
-            # get the unique sequences and compare the length with all sequences
-            unique_seq = list(set(list(allele_seq.values())))
-            if len(unique_seq) < len(allele_seq):
-                tmp_dict = {}
-                for rec_id, seq_value in allele_seq.items():
-                    if seq_value not in tmp_dict:
-                        tmp_dict[seq_value] = 0
-                    else:
-                        bad_quality_record.append(rec_id)
-                        a_quality[rec_id]["quality"] ="Bad quality"
-                        a_quality[rec_id]["reason"] ="Duplicate allele"
-        if self.remove_subset:
-            unique_seq = list(set(list(allele_seq.values())))
+        # check if there are duplicated alleles
+        # get the unique sequences and compare the length with all sequences
+        unique_seq = list(set(list(allele_seq.values())))
+        if len(unique_seq) < len(allele_seq):
+            tmp_dict = {}
             for rec_id, seq_value in allele_seq.items():
-                unique_seq.remove(seq_value)
-                if seq_value in unique_seq:
+                if seq_value not in tmp_dict:
+                    tmp_dict[seq_value] = 0
+                else:
+                    a_quality[rec_id]["quality"] = "Bad quality"
+                    a_quality[rec_id]["reason"] = "Duplicate allele"
+                    if self.remove_duplicated:
+                        bad_quality_record.append(rec_id)
+
+        for rec_id, seq_value in allele_seq.items():
```

Comment:
can't you include this check above so we only iterate one time through the dict? 
Maybe changing the if clause position?

Reply:
I can not as I need to remove it, for including in the bad quality record , once I know that it is duplicated